bpf/bpf_lxc.c

```diff
@@ -1048,9 +1065,41 @@ ipv4_policy(struct __sk_buff *skb, int ifindex, __u32 src_label, __u8 *reason, _
 		verdict = 0;
 
 	if (ret == CT_NEW) {
+#ifdef ENABLE_DSR
+		// Check whether IPv4 header contains a 64-bit option (IPv4 header
+		// w/o option (5 x 32-bit words) + the DSR option (2 x 32-bit words))
+		if (ip4->ihl == 0x7) {
```

Comment:
What happens of the IPv4 packet carries already other options aside our custom one? Why ignore DSR in that case (seems unexpected)?

Reply:
Note that this will turn off GRO at the server side.

Reply:
> What happens if the IPv4 packet carries already other options aside our custom one

I didn't want to increase verifier complexity by adding a proper IPv4 options parser. I will document this, and implement as a follow-up.

> Note that this will turn off GRO at the server side

Good point, thanks for noting.